sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/CacheTableExec.scala

```diff
@@ -87,3 +87,16 @@ case class CacheTableAsSelectExec(
     sparkSession.table(tempViewName)
   }
 }
+
+case class UncacheTableExec(
+    relation: LogicalPlan,
+    cascade: Boolean) extends V2CommandExec {
+  override def run(): Seq[InternalRow] = {
+    val sparkSession = sqlContext.sparkSession
+    val df = Dataset.ofRows(sparkSession, relation)
+    sparkSession.sharedState.cacheManager.uncacheQuery(df, cascade)
```

Comment:
`uncacheQuery` can take `LogicalPlan` directly. Let's use that overload to avoid creating a dataframe.

Reply:
We can do the same to `cacheQuery`, but we need to add a new overload that takes `LogicalPlan` first.

Reply:
I updated to pass a logical plan instead of dataframe (this required updating more rules, but I think it's "more correct".)

I will add a new overload that takes `LogicalPlan` in a separate PR.